Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift

```diff
@@ -8,64 +8,44 @@
 
 import Foundation
 
-// TODO: refactor key with enum key type.
-func storedKey(_ key: String, serviceType: ServiceType) -> String {
+func storedKey(_ key: StoredKey, serviceType: ServiceType) -> String {
     // This key should be compatible with existing OpenAI config keys
     // EZOpenAIServiceUsageStatusKey
     // EZOpenAIDictionaryKey
-    "EZ" + serviceType.rawValue + key + "Key"
-}
-
-func serviceUsageStatusStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZServiceUsageStatusKey, serviceType: serviceType)
-}
-
-func translationStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZTranslationKey, serviceType: serviceType)
-}
-
-func sentenceStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZSentenceKey, serviceType: serviceType)
-}
-
-func dictionaryStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZDictionaryKey, serviceType: serviceType)
-}
-
-func availableModelsStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZAvailableModelsKey, serviceType: serviceType)
-}
-
-func validModelsStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZValidModelsKey, serviceType: serviceType)
-}
-
-func modelStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZModelKey, serviceType: serviceType)
-}
-
-func apiStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZAPIKey, serviceType: serviceType)
-}
-
-func endpointStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZEndpointKey, serviceType: serviceType)
-}
-
-func nameStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZNameKey, serviceType: serviceType)
+    "EZ" + serviceType.rawValue + key.rawValue.capitalizeFirstLetter() + "Key"
 }
 
 extension UserDefaults {
-    static func bool(forKey key: String, serviceType: ServiceType) -> Bool {
+    static func bool(forKey key: StoredKey, serviceType: ServiceType) -> Bool {
         let key = storedKey(key, serviceType: serviceType)
         let value = standard.bool(forKey: key)
         return value
     }
 
-    static func string(forKey key: String, serviceType: ServiceType) -> String? {
+    static func string(forKey key: StoredKey, serviceType: ServiceType) -> String? {
         let key = storedKey(key, serviceType: serviceType)
         let value = standard.string(forKey: key)
         return value
     }
 }
+
+// MARK: - StoredKey
+
+enum StoredKey: String {
+    case serviceUsageStatus
+    case translation
+    case dictionary
+    case sentence
+    case supportedModels = "AvailableModels" // save in String: "gpt-3.5, gpt-4"
```

Comment:
The default string will be lowered case, maybe make them consistent here?

Reply:
The key here needs to be compatible with the previous constants `EZOpenAIAvailableModelsKey` and `EZCustomOpenAIAvailableModelsKey`. 

Do you have any other ideas, please give details.

Reply:
OK, got it, this should be fine.